tests/_support/Helper/Config/Environment.php

```
@@ -0,0 +1,45 @@
+<?php
+
+
+namespace Helper\Config;
+
+
+use PHPUnit\Exception;
+
+/**
+ * Class Environment
+ * @package Helper\Config
+ */
+class Environment
+{
+
+    /**
+     * @var
+     */
+    private $env;
+
+    private $requiredEnvVariables = ['SHOP_SYSTEM', 'SHOP_SYSTEM_CONTAINER_NAME'];
+
+    /**
+     * Environment constructor.
+     */
+    public function __construct()
+    {
+        $environment = $_ENV;
+        foreach ( $this->requiredEnvVariables as $var) {
+            try {
+                $this->env[$var] = $environment[$var];
+            } catch (Exception $e) {
```

I would throw the exceptions in the constructor.

My rationale goes like this: if these parameters are required, then it's better to fail early and not allow an Environment object to exist in the system at all, than to have an Environment object which is malfunctioning.

Others parts of the system (method, classes) that also require an Environment will not be callable, because you don't have a valid $environment to give them.